open-hackathon-adminUI/src/app/database/models.py

```diff
@@ -0,0 +1,161 @@
+import sys
+import email
+import os
+
+sys.path.append("..")
+from . import UserMixin
+from . import db
+from datetime import datetime
+import uuid
+import json
+
+
+def to_json(inst, cls):
+    # add your coversions for things like datetime's
+    # and what-not that aren't serializable.
+    convert = dict()
+    convert[db.DateTime] = str
+
+    d = dict()
+    for c in cls.__table__.columns:
+        v = getattr(inst, c.name)
+        if c.type.__class__ in convert.keys() and v is not None:
+            try:
+                func = convert[c.type.__class__]
+                d[c.name] = func(v)
+            except:
+                d[c.name] = "Error:  Failed to covert using ", str(convert[c.type.__class__])
+        elif v is None:
+            d[c.name] = str()
+        else:
+            d[c.name] = v
+    return json.dumps(d)
+
+
+
+class AdminUser(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(50))
+    nickname = db.Column(db.String(50))
+    openid = db.Column(db.String(100))
+    avatar_url = db.Column(db.String(200))
+    access_token = db.Column(db.String(100))
+    online = db.Column(db.Integer)  # 0:offline 1:online
+    create_time = db.Column(db.DateTime)
+    last_login_time = db.Column(db.DateTime)
+
+    def get_admin_user_id(self):
+        return self.id
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, **kwargs):
+        super(AdminUser, self).__init__(**kwargs)
+        if self.create_time is None:
+            self.create_time = datetime.utcnow()
+        if self.last_login_time is None:
+            self.last_login_time = datetime.utcnow()
+#        if self.slug is None:
+#            self.slug = str(uuid.uuid1())[0:8]  # todo generate a real slug
+
+    def __repr__(self):
+        return "AdminUser: " + self.json()
+
+    def is_authenticated(self):
+        return True
+
+    def is_active(self):
+        return True
+
+    def is_anonymous(self):
+        return False
+
+    def get_id(self):
+        return unicode(self.id)
+
```

Comment:
the base class of `AdminUser` is `UserMixin` which  also defines the four methods `get_id`,`is_anonymous`,`is_authenticated`.